modules/gapi/include/opencv2/gapi/imgproc.hpp

```diff
@@ -507,7 +507,7 @@ namespace imgproc {
                 int outSz_w = saturate_cast<int>(in.size.width  * fx);
                 int outSz_h = saturate_cast<int>(in.size.height * fy);
                 GAPI_Assert(outSz_w > 0 && outSz_h > 0);
-                return in.withSize(Size(outSz_w, outSz_h));
+                return in.withType(in.depth, in.chan).withSize(Size(outSz_w, outSz_h));
```

Comment:
why do we need to apply `withType`?
also we get type from `in` and apply it to `in` back - how is it supposed to be changed?

Reply:
It speak that type of output matrix must be the same as input type.

Reply:
as i see you returns copy of `in` which have type of `in` already without additional `withType`

Reply:
In GAPI_TYPED_KERNEL we always sort of return a copy of the input. These are features of the implementation of the graph. However, we must make sure that the output type matches our requirements. And if this is not the case for a particular case, then such a graph will not pass the test, since the test function will swear at our attempt.

Reply:
Ok. Applied.